subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/UpgradeDependencyPlugin.java

```diff
@@ -198,6 +198,8 @@ public void execute(final CreatePullRequestTask task) {
                 task.setAuthToken(conf.getLenient().getGitHub().getWriteAuthToken());
                 task.setVersionBranch(getVersionBranchName(upgradeDependencyExtension));
                 task.setVersionUpgrade(upgradeDependencyExtension);
+                task.setPullRequestTitle(conf.getGit().getPullRequestTitle());
+                task.setPullRequestDescription(conf.getGit().getPullRequestDescription());
```

Comment:
Since this configuration of **CreatePullRequest** task is used specifically for upgrading dependencies, we can use default values that we had previously hardcoded in **CreatePullRequest** class. So for **pullRequestTitle** it would be "Version of %s upgraded to %s". 

This way user can use the default value or override it like that:

apply plugin: "org.shipkit.upgrade-dependency"
project.tasks.createPullRequest.pullRequestTitle = "some title"

Reply:
This would be more complicated. The scope of this issue is only to allow task to be configurable. No need for complex smart defaults, just the default specified in the plugin at the moment would suffice. Since 'pullRequestTitle' and 'pullRequestDescription' will be now configurable user will still be able to set it to anything they want.